From fc272948fbc4a31ecb1da3914138a32033953ec0 Mon Sep 17 00:00:00 2001
From: DongHwan Yu <donghwan.yu@lge.com>
Date: Mon, 11 Nov 2024 15:14:20 +0900
Subject: [PATCH] [build] 132.0.6815.0~1: Disable Skia Rust-based Fontations

:Release Notes:
Disabled Skia Rust-based Fontations to fix build error
after rolling to 132.0.6815.0~1

:Detailed Notes:
In upstream skia CL https://skia-review.googlesource.com/c/skia/+/914656
relanded "Reland "Reland "Adding Fontations to FontConfig manager"""
based on Rust, which is not yet supported by Neva. So this patch
disabled it.

Without this change the build fails with configuration errors (and also
linker errors):
  In file included from ../../third_party/skia/src/ports/SkFontScanner_fontations.cpp:8:
  ../../third_party/skia/src/ports/SkTypeface_fontations_priv.h:20:10:
  fatal error: src/ports/fontations/src/ffi.rs.h: No such file or directory
    20 | #include "src/ports/fontations/src/ffi.rs.h"
       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

:Issues Addressed:
[NEVA-9746] Upgrade neva-main-rolling up to 6815 BUILD
---
 gn/ports.gni | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/gn/ports.gni b/gn/ports.gni
index 7b97859680..7e166c21f3 100644
--- a/gn/ports.gni
+++ b/gn/ports.gni
@@ -7,6 +7,8 @@
 
 # To update this file, run make -C bazel generate_gni
 
+import("//neva/neva.gni")
+
 _src = get_path_info("../src", "abspath")
 _include = get_path_info("../include", "abspath")
 
@@ -27,6 +29,10 @@ skia_ports_freetype_sources = [
   "$_src/ports/SkTypeface_FreeType.h",
 ]
 
+if (use_neva_appruntime) {
+  skia_ports_freetype_sources -= [ "$_src/ports/SkFontScanner_FreeType_priv.h" ]
+}
+
 # Generated by Bazel rule //include/ports:android_fontmgr_hdrs
 skia_ports_fontmgr_android_public = [ "$_include/ports/SkFontMgr_android.h" ]
 
@@ -124,6 +130,13 @@ skia_ports_typeface_fontations_sources = [
   "$_src/ports/SkTypeface_fontations_priv.h",
 ]
 
+if (use_neva_appruntime) {
+  skia_ports_typeface_fontations_sources -= [
+    "$_src/ports/SkFontScanner_fontations.cpp",
+    "$_src/ports/SkFontScanner_fontations_priv.h",
+  ]
+}
+
 # Generated by Bazel rule //src/ports:osfile_win
 skia_ports_windows_sources = [ "$_src/ports/SkOSFile_win.cpp" ]
 
-- 
2.45.2

